test/unit/processing/components/get-schedule-dates.test.js

```diff
@@ -93,47 +93,38 @@ describe('get schedule dates', () => {
     expect(schedule[3].value).toEqual('500.00')
   })
 
-  test('should insert top up adjustment after last outstanding payment in original schedule', () => {
+  test('should insert top up adjustment immediate payment as first item of schedule', () => {
```

Comment:
This test name isn't descriptive enough. Or it relies on A LOT of known content.

So from this I deduce an immediate payment will only have a period set to the current date in given format.

Nothing about its `paymentType`, `period`, `value`, etc .

Ofc that's not true as the test below is stating what the `period` will be but it relies on these tests being 'connected'. Tests should be independent.

We could bundle these all into 1 test but tht's not good practise either.

I think having these sub-describes will make it much easier to know that these tests are connected due to a common scenario, 'when a top-up'.

Do think the test names need re-evaluated